controllers/nutanixcluster_controller.go

```diff
@@ -95,10 +95,42 @@ func (r *NutanixClusterReconciler) SetupWithManager(ctx context.Context, mgr ctr
 			),
 			builder.WithPredicates(predicates.ClusterUnpausedAndInfrastructureReady(ctrl.LoggerFrom(ctx))),
 		).
+		Watches(
+			&infrav1.NutanixFailureDomainV2{},
+			handler.EnqueueRequestsFromMapFunc(
+				r.mapNutanixFailureDomainV2ToNutanixCluster(),
+			),
+		).
 		WithOptions(copts).
 		Complete(r)
 }
 
+func (r *NutanixClusterReconciler) mapNutanixFailureDomainV2ToNutanixCluster() handler.MapFunc {
```

Comment:
Don't we want to only return clusters that are using the failure domain? 

Also do you think we should add a fieldIndexer here?

Reply:
Fixed.